plugins/plugin.go

```diff
@@ -263,7 +266,7 @@ func makeMetricsRequest(client *http.Client, request *http.Request) (producers.M
 
 	// 204 No Content is not an error code; we handle it explicitly
 	if resp.StatusCode == http.StatusNoContent {
-		l.Warnf("Empty response received from endpoint: %+v", request.URL)
+		l.Warnf("Empty response received from endpoint: %+v", url)
```

Comment:
We can just use `%s` here for formatting since it's a string.  Same with line 252 above.